solr/core/src/test/org/apache/solr/core/CoreSorterTest.java

```diff
@@ -71,168 +69,141 @@ public void testComparator() {
           }
         }
       }
 
-      for (Map.Entry<ReplicaInfo, String> e : replicaPositions.entrySet()) {
-        if (e.getValue().equals(myNodeName)) {
-          myCores.add(e.getKey().getCloudDescriptor());
-          localCores.add(new MockCoreContainer.MockCoreDescriptor() {
-            @Override
-            public CloudDescriptor getCloudDescriptor() {
-              return e.getKey().getCloudDescriptor();
-            }
-          });
-        }
-      }
-    }
-
-    @Override
-    String getNodeName() {
-      return myNodeName;
-    }
-
-    @Override
-    Collection<CloudDescriptor> getCloudDescriptors() {
-      return myCores;
-
-    }
+      NodeConfig mockNodeConfig = mock(NodeConfig.class);
+      when(mockNodeConfig.getNodeName()).thenReturn(thisNode);
+      when(mockCC.getNodeConfig()).thenReturn(mockNodeConfig);
 
-    public List<CoreDescriptor> getLocalCores() {
-      return localCores;
     }
 
-    @Override
-    Collection<Replica> getReplicas(ClusterState cs, String coll, String slice) {
-      List<Replica> r = new ArrayList<>();
-      for (Map.Entry<ReplicaInfo, String> e : replicaPositions.entrySet()) {
-        if (e.getKey().equals(coll, slice)) {
-          r.add(e.getKey().getReplica(e.getValue()));
+    List<CoreDescriptor> myDescs = new ArrayList<>(myDescsToCounts.keySet());
+    for (int i = 0; i < 10; i++) {
+      Collections.shuffle(myDescs, random());
+
+      List<CoreDescriptor> resultDescs = CoreSorter.sortCores(mockCC, myDescs);
+      // map descriptors back to counts, removing consecutive duplicates
+      List<CountsForEachShard> resultCounts = new ArrayList<>();
+      CountsForEachShard lastCounts = null;
+      for (CoreDescriptor resultDesc : resultDescs) {
+        CountsForEachShard counts = myDescsToCounts.get(resultDesc);
+        if (counts != lastCounts) {
+          resultCounts.add(counts);
         }
+        lastCounts = counts;
       }
-      return r;
+      assertEquals(expectedCounts, resultCounts);
     }
   }
 
+  private CoreDescriptor newCoreDescriptor(Replica r) {
+    Map<String,String> props = map(
+        CoreDescriptor.CORE_SHARD, r.getSlice(),
+        CoreDescriptor.CORE_COLLECTION, r.getCollection(),
+        CoreDescriptor.CORE_NODE_NAME, r.getNodeName()
+    );
+    return new CoreDescriptor(r.getCoreName(), TEST_PATH(), props , null, mock(ZkController.class));
+  }
+
+  protected Replica addNewReplica(List<Replica> replicaList, String collection, String slice, List<String> possibleNodes) {
+    String replica = "r" + replicaList.size();
+    String node = possibleNodes.get(random().nextInt(possibleNodes.size())); // place on a random node
+    Replica r = newReplica(collection, slice, replica, node);
+    replicaList.add(r);
+    return r;
+  }
+
+  protected Replica newReplica(String collection, String slice, String replica, String node) {
+    return new Replica(replica, Utils.makeMap("core", replica, "node_name", node), collection, slice);
```

Comment:
`s/Utils.makeMap/map/`?

Reply:
That'd work too; I took this bit here from an existing line of code in the previous version of this test.